### ISSUE TITLE
[Bug]: Can't add new kanban cards after markdown/kanban switch bug hit

### ISSUE THREAD

**Issue by OogieM** [mgmeyers/obsidian-kanban#399]
### Describe the bug

I can use a kanban board for quite a while but at some point I can no longer add cards to it. This happens after I've run into the bug with changes made in markdown mode not showing up. Once I hit that I can no longer add cards to that board no matter what mode I am in.

If I close Obsidian down and reopen same behavior.

### Expected Behavior

No matter how many times if flip in and out of markdown vs kanban mode I should always see all cards and be able to edit all cards. 

### Steps to reproduce

I have no idea what I do that starts this but it seems to happen only after I've seen the issue of edits made in markdown not being displayed in kanban mode. 

So far the only solution has been to create a new kanban board, and hand copy all the old cards into new one. Obviously this won't work long term. 

I did map option k to be a hot key that toggles between kanban and markdown mode

### Relevant errors (if available)

```shell
Loaded Charts View plugin
main.ts:85 Dataview: Version 0.4.21 Loaded (Compiled November 20, 2021 14:41:21)
main.ts:10 Loading Hotkeys++ plugin
VM109:16 Kindle Highlights plugin: loading plugin 12/4/2021, 7:21:44 AM
DevTools failed to load SourceMap: Could not load content for app://obsidian.md/main.js.map: 
VM109:16 loading obsidian-tracker plugin
VM109:16 loading smart-random-note
VM109:16 loading plugin: Obsidian Charts
17[Violation] Added non-passive event listener to a scroll-blocking <some> event. Consider marking event handler as 'passive' to make the page more responsive. See <URL>
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
on @ codemirror.js:536
registerEventHandlers @ codemirror.js:7994
CodeMirror @ codemirror.js:7919
CodeMirror @ codemirror.js:7878
e @ app.js:1
t @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.setViewState @ app.js:1
eval @ VM113:74
o @ VM113:23
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.deserializeLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.deserializeLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.setLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
s @ app.js:1
Show 9 more frames
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchmove' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
on @ codemirror.js:536
registerEventHandlers @ codemirror.js:8007
CodeMirror @ codemirror.js:7919
CodeMirror @ codemirror.js:7878
e @ app.js:1
t @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.setViewState @ app.js:1
eval @ VM113:74
o @ VM113:23
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.deserializeLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.deserializeLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.setLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
s @ app.js:1
Show 9 more frames
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'mousewheel' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
on @ codemirror.js:536
registerEventHandlers @ codemirror.js:8040
CodeMirror @ codemirror.js:7919
CodeMirror @ codemirror.js:7878
e @ app.js:1
t @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.setViewState @ app.js:1
eval @ VM113:74
o @ VM113:23
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.deserializeLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.deserializeLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.setLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
s @ app.js:1
Show 9 more frames
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
on @ codemirror.js:536
registerEventHandlers @ codemirror.js:7994
CodeMirror @ codemirror.js:7919
CodeMirror @ codemirror.js:7878
e @ app.js:1
t @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.setViewState @ app.js:1
eval @ VM113:74
o @ VM113:23
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.deserializeLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
s @ app.js:1
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchmove' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
on @ codemirror.js:536
registerEventHandlers @ codemirror.js:8007
CodeMirror @ codemirror.js:7919
CodeMirror @ codemirror.js:7878
e @ app.js:1
t @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.setViewState @ app.js:1
eval @ VM113:74
o @ VM113:23
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.deserializeLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
s @ app.js:1
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'mousewheel' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
on @ codemirror.js:536
registerEventHandlers @ codemirror.js:8040
CodeMirror @ codemirror.js:7919
CodeMirror @ codemirror.js:7878
e @ app.js:1
t @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.setViewState @ app.js:1
eval @ VM113:74
o @ VM113:23
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.deserializeLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
s @ app.js:1
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
eval @ VM113:27
dl @ VM113:23
eval @ VM113:23
zy @ VM113:23
e @ VM109:333
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
eval @ VM113:27
dl @ VM113:23
eval @ VM113:23
zy @ VM113:23
e @ VM109:333
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
eval @ VM113:27
dl @ VM113:23
eval @ VM113:23
zy @ VM113:23
e @ VM109:333
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
eval @ VM113:27
dl @ VM113:23
eval @ VM113:23
zy @ VM113:23
e @ VM109:333
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
eval @ VM113:27
dl @ VM113:23
eval @ VM113:23
zy @ VM113:23
e @ VM109:333
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
eval @ VM113:27
dl @ VM113:23
eval @ VM113:23
zy @ VM113:23
e @ VM109:333
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
eval @ VM113:27
dl @ VM113:23
eval @ VM113:23
zy @ VM113:23
e @ VM109:333
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
eval @ VM113:27
dl @ VM113:23
eval @ VM113:23
zy @ VM113:23
e @ VM109:333
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
eval @ VM113:27
dl @ VM113:23
eval @ VM113:23
zy @ VM113:23
e @ VM109:333
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
eval @ VM113:27
dl @ VM113:23
eval @ VM113:23
zy @ VM113:23
e @ VM109:333
VM109:333 [Violation] Added non-passive event listener to a scroll-blocking 'wheel' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952
eval @ VM109:333
eval @ VM109:16
e @ app.js:1
t @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.setViewState @ app.js:1
eval @ VM113:74
o @ VM113:23
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.deserializeLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
o @ app.js:1
t.deserializeLayout @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
s @ app.js:1
VM109:16 Dataview: Task & metadata parsing queued in 0.083s.
VM109:16 Dataview: File prefix tree built in 0.004s.
[Violation] Forced reflow while executing JavaScript took 30ms
[Violation] 'setTimeout' handler took 117ms
[Violation] Forced reflow while executing JavaScript took 45ms
[Violation] 'requestAnimationFrame' handler took 88ms
```

### Operating system

macOS

**Comment by mgmeyers** [comment]
Thanks for reporting this @OogieM. I'll have to dig around and see what could be causing it.

**Comment by OogieM** [comment]
A bit more information that may or may not be relevant. I think (not done extensive testing yet) that I can force the errors when I by force of habit try to enter markdown text as I am entereing in a card in kanban view. I always want the card title in bold and my fingers just naturally type **title** even when entering in in the kanban mode. That seems to be a trigger for weird things to start happening. The bold doesn't show so I go into markdown and redo it there. As I said not sure but I've noticed that i do that a lot so it may be important.

**Comment by mgmeyers** [comment]
@OogieM Oh, interesting. Could you write out the steps you take that ends up with the bold not showing? Even better if you could record a video of it in action.

**Comment by gernotstarke** [comment]
similar issue here: I switched between kanban- and markdown mode a few times, and bang... all cards have vanished in k-mode, whereas the markdown still looks consistent to me.

**Comment by OogieM** [comment]
OK I can't seem to capture a video. I never know when this is going to occur to start it before it happens!

However part of it may be related to whether you use a return to create a new card or be a newline in the card. I have my system set so that to add a newline in a card is shift + enter and plain newline creates a card. I use Omnifocus too and its normal mode is that an enter will create a new task or a new project. So that's what's in my muscle memory for a new thing. Not sure if that helps or not but it is new info on how I use the plug-in that may be relevant.

**Comment by gernotstarke** [comment]
With Obsidian 0.12.19: when uninstalling the advanced-tables plugin, the kanban view re-appears...

**Comment by OogieM** [comment]
I don't use the Advanced Tables plug-in or have it installed so that doesn't solve my issue.

**Comment by mgmeyers** [comment]
So, I've spent quite a bit of time trying to reproduce this issue with no luck. I have seen it in the past, so I know it's a thing, but for whatever reason I can't trigger it intentionally. I've added a bunch of new error handling code that, at the very least, should prevent any errors from sneaking by unhandled. What I think happens is that the plugin gets into some sort of error state and de-syncs from the appropriate markdown state.

Let me know if you notice any new errors when you run into this issue. I'll keep on testing, though, to see if I can hunt down what's going on.

**Comment by gernotstarke** [comment]
Thanx for your effort!

Dr. Gernot Starke
+49-177-7282570
***@***.***
www.gernotstarke.de

Von einem mobilen Geraet gesendet - bitte verzeihen Sie eventuelle Tippfehler.

**Comment by OogieM** [comment]
ok will do, thanks for working on it.

**Comment by ValHeyrie** [comment]
Hello ! 
(I'm French)

Similar issue here ?

A new list is automatically created after I click on a link to another note; in a note in my kanban. 
And the kanban doesn't work anymore; it's frozen. 

if I switch to markdown mode and then reopen the kanban board; the kanban disappears. 

I can fix these 2 problems by deactivating, then reactivating Kanban in the settings; then opening a kanban board on the top right. But it's annoying in the long run. 

Translated with www.DeepL.com/Translator (free version)

**Comment by mgmeyers** [comment]
Hey all, I fixed another bug that's similar to this one and it may have fixed this one as will. Could you let me know if you still run into this issue with the latest version of the Kanban plugin?

**Comment by gernotstarke** [comment]
Thanx for caring! It's difficult to tell if it's really fixed as it occurred "randomly" - but the plugin seems to be stable currently. I switched between card- and markdown view a dozen or more times, all ok. (and in the meantime convinced several other obsidian users to use your Kanban plugin too!)

**Comment by mgmeyers** [comment]
Hey all, I'm wondering if any of you have run into this issue lately? I think I may have fixed it a while back, but because I wasn't able to reproduce, I'm not sure. Let me know!

**Comment by OogieM** [comment]
So far for me it's all working correctly. I did change my preferences so enter ends the card rather than the shift + enter and that may actually have been a few of the issues for me. I do have to save them as cards before changing modes or they do get lost. You have to actually have finished making a card.

**Comment by mgmeyers** [comment]
Ok, I'm going to close this for now. Let me know if you still run into issues with it!

**Comment by jwhitley** [comment]
Ack, sorry, I did also try to repro (this week's current desktop and iOS insider builds) and everything worked great... then I forgot to report back. So yes, confirmed as resolved here. 🎊
